koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -330,27 +330,63 @@ def _incremental_upload_logs(self, child_pid):
         finally:
             watcher.clean()
 
-    def _write_incremental_logs(self, build_id, log_filename):
-        log_basename = os.path.basename(log_filename)
+    def _write_combined_log(self, log, logs_dir):
+        log_basename = 'openshift-incremental.log'
+        log_filename = os.path.join(logs_dir, log_basename)
+
         self.logger.info("Will write follow log: %s", log_basename)
-        try:
-            build_logs = self.osbs().get_build_logs(build_id,
-                                                    follow=True)
-        except Exception, error:
-            msg = "Exception while waiting for build logs: %s" % error
-            raise ContainerError(msg)
         outfile = open(log_filename, 'w')
         try:
-            for line in build_logs:
+            for line in log:
                 outfile.write("%s\n" % line)
-                outfile.flush()
+            outfile.flush()
         except Exception, error:
             msg = "Exception (%s) while reading build logs: %s" % (type(error),
                                                                    error)
             raise ContainerError(msg)
         finally:
             outfile.close()
         self.logger.info("%s written", log_basename)
+
+    def _write_demultiplexed_logs(self, logs, logs_dir):
+        self.logger.info("Will write demuxed logs in: %s/", logs_dir)
+        platform_logs = {}
+        for entry in logs:
+            platform = entry.platform
+            if platform not in platform_logs:
+                prefix = 'orchestrator' if platform is None else platform
+                log_filename = os.path.join(logs_dir, "%s.log" % prefix)
+                platform_logs[platform] = open(log_filename, 'w')
+            try:
+                platform_logs[platform].write((entry.line + '\n').encode('utf-8'))
+            except Exception, error:
+                msg = "Exception (%s) while reading build logs: %s" % (type(error),
+                                                                       error)
```

Comment:
While reading? Should that say "writing"?

Reply:
Ah, probably. Copy-n-paste error mirroring the (probably also incorrect) code that was already there for writing out openshift-incremental.log.